lib/server-builder.js

```diff
@@ -56,25 +66,40 @@ class ServerBuilder {
       await this.insertAdditionalAppDependencies(this.additionalAppDependencies);
     }
 
-    await this.runBuildProcess();
+    await this.runBuildProcess(testAppPackage, signingConfig);
   }
 
-  getCommand () {
+  getCommand (testAppPackage, signingConfig) {
     const cmd = system.isWindows() ? 'gradlew.bat' : './gradlew';
     let args = VERSION_KEYS
       .filter(key => key !== GRADLE_VERSION_KEY)
       .map(key => {
         const serverVersion = this.serverVersions[key];
         const gradleProperty = `appium${key.charAt(0).toUpperCase()}${key.slice(1)}`;
-        return serverVersion ? `-P${gradleProperty}=${serverVersion}` : null;
+        return this.buildGradleProperty(gradleProperty, serverVersion);
       })
       .filter(Boolean);
 
+    if (signingConfig) {
+      args.push(this.buildGradleProperty('appiumZipAlign', signingConfig.zipAlign));
+      args.push(this.buildGradleProperty('appiumKeystoreFile', signingConfig.keystoreFile));
+      args.push(this.buildGradleProperty('appiumKeystorePassword', signingConfig.keystorePassword));
+      args.push(this.buildGradleProperty('appiumKeyAlias', signingConfig.keyAlias));
+      args.push(this.buildGradleProperty('appiumKeyPassword', signingConfig.keyPassword));
+    }
+
+    if (testAppPackage) {
+      args.push(this.buildGradleProperty('appiumTargetPackage', testAppPackage));
+    }
     args.push('assembleAndroidTest');
 
     return {cmd, args};
   }
 
+  buildGradleProperty (key, value) {
```

Comment:
we don't need a separate method for this helper. This could be rather an external function or a nested function (it is anyway only used in the single method)